File: backend/src/models/conversation.py

from sqlalchemy import String, Boolean, DateTime, Text, ForeignKey
from sqlalchemy.orm import Mapped, mapped_column
from sqlalchemy.sql import func
from datetime import datetime
import uuid
from ..core.database import Base

class Conversation(Base):
    __tablename__ = "conversations"
    
    id: Mapped[str] = mapped_column(String(36), primary_key=True, default=lambda: str(uuid.uuid4()))
    user_id: Mapped[str] = mapped_column(String(36), ForeignKey("users.id", ondelete="CASCADE"), index=True, nullable=False)
    project_id: Mapped[str] = mapped_column(String(36), ForeignKey("projects.id", ondelete="SET NULL"), nullable=True)
    title: Mapped[str] = mapped_column(String(500), nullable=False, default="New Chat")
    archived: Mapped[bool] = mapped_column(Boolean, default=False)
    meta_data: Mapped[str] = mapped_column(Text, default="{}")  # JSON as text for simplicity, will be JSONB in PG
    created_at: Mapped[datetime] = mapped_column(DateTime(timezone=True), server_default=func.now())
    updated_at: Mapped[datetime] = mapped_column(DateTime(timezone=True), server_default=func.now(), onupdate=func.now())

File: lib/features/ai_chat/domain/entities/conversation.dart
class Conversation {
  final String id;
  final String userId;
  final String? projectId;
  final String title;
  final DateTime createdAt;
  final DateTime updatedAt;
  final bool archived;
  final Map<String, dynamic> metadata;

  const Conversation({
    required this.id,
    required this.userId,
    this.projectId,
    required this.title,
    required this.createdAt,
    required this.updatedAt,
    this.archived = false,
    this.metadata = const {},
  });

  Conversation copyWith({
    String? id,
    String? userId,
    String? projectId,
    String? title,
    DateTime? createdAt,
    DateTime? updatedAt,
    bool? archived,
    Map<String, dynamic>? metadata,
  }) {
    return Conversation(
      id: id ?? this.id,
      userId: userId ?? this.userId,
      projectId: projectId ?? this.projectId,
      title: title ?? this.title,
      createdAt: createdAt ?? this.createdAt,
      updatedAt: updatedAt ?? this.updatedAt,
      archived: archived ?? this.archived,
      metadata: metadata ?? this.metadata,
    );
  }
}
